### DISCUSSION TITLE
gh-pages fails to render images from public GitHub repo (403 forbidden)

### DISCUSSION THREAD

**Discussion by ljmoody** [community/community#137093] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I'm trying to include images on a GitHub pages site. I'm using an `<img>` tag and for the src attribute, I am providing a link to the image in our public GitHub repo with "?raw=true" at the end to make sure I am only getting the image and not the html of a web page. 

For most of the `<img>` tags on my GitHub pages site, the network request is redirected, but resolves with a code of 302. For those images, they show up on the GitHub pages site correctly. Some of the images however, are failing to load properly. In the network panel, I am seeing that the requests for these images are resolving in a 403 Forbidden error. 

This confuses me as all of the images are from public GitHub repos. I double checked by accessing the image URL in a incognito Chrome window, without signing in to GitHub, and I was able to see the image there.
 
Any thoughts on where to go from here in attempting to uncover the cause of this issue?

**Comment by theshawnv** [comment]
Your situation is indeed puzzling 🤔, especially since the images are publicly accessible and some work while others do not. There are a few potential reasons why you might be encountering 403 Forbidden errors.
Here are some things you could investigate:

1.Rate Limiting, 2.Case Sensitivity in File Names, and 3.Cache and Network Issues

**Comment by ljmoody** [comment, marked as answer]
Hi @shayan0v0n, I appreciate your input and suggestions!
This has been a puzzling one for me and my team. I had previously investigated the case sensitivity and cache/network issues per team recommendations and other discussion thread suggestions. I was researching rate limiting when a team member had an idea...he remembered inspecting the web page's network activity and seeing 200, 302, and 403 status for images. 

Since others may find this useful for their own Pages project, I'm sharing our findings and solution to this issue:
- The image syntax I was using: 
`<img alt="An image of our registration page." src="https://github.com/OWNER/REPO/blob/gh-pages/images/IMAGE_FILE_NAME.png?raw=true">`
- @matthewtomo noted that, while the source code we used did work and displayed the image (not the html of a web page), the URL for all intact images was redirected to
`https://raw.githubusercontent.com/OWNER/REPO/gh-pages/images/IMAGE_FILE_NAME.png`
- The hypothesis: redirects may be causing image display & 403 issues. When too many redirects happen (there were 113 images in one .md file), the images display (or not) inconsistently. 
- We tested by replacing all syntax with this:
`<img alt="Image of IMAGE DESCRIPTION." src="https://raw.githubusercontent.com/OWNER/REPO/gh-pages/images/IMAGE_FILE_NAME.png>`

**ET VOILÀ! -- PROBLEM SOLVED!**
- The syntax I was using originally with `?raw=true` _may_ work for Pages sites without some magic upper limit of images on them, but because that part of the syntax was causing a multitude of redirects, it was the source of our image display issue.
